system/db/content.go

```diff
@@ -36,6 +36,98 @@ func SetContent(target string, data url.Values) (int, error) {
 	return update(ns, id, data)
 }
 
+// UpdateContent inserts or updates values in the database.
+// Updated content will be merged into existing values from the database.
+// The `target` argument is a string made up of namespace:id (string:int)
+func UpdateContent(target string, data url.Values) (int, error) {
+	t := strings.Split(target, ":")
+	ns, id := t[0], t[1]
+
+	// check if content id == -1 (indicating new post).
+	// if so, run an insert which will assign the next auto incremented int.
+	// this is done because boltdb begins its bucket auto increment value at 0,
+	// which is the zero-value of an int in the Item struct field for ID.
+	// this is a problem when the original first post (with auto ID = 0) gets
+	// overwritten by any new post, originally having no ID, defauting to 0.
+	if id == "-1" {
```

Comment:
>Agree that a "-1" id should be invalid for updates, right now updateContentHandler to rejects requests with a missing ID and the handler could reject id="-1" also.

Instead of rejecting a "-1" id, we could just do a type assertion for `Externalable` (or `Addable` if we go the route of renaming), which would then let us use the same SetContent|UpdateContent func. 

 >There really isn't much different between SetContent and the added UpdateContent other than merging content into existing values. I wonder if you really still need the old behavior of SetContent and the existing update function with 'replace' behavior or if the merge behavior could work for everything? If the "-1" validation is in the handler, we might be able to eliminate some duplication here and just replace SetContent with UpdateContent...

I agree they are similar and think duplication like this could lead to bugs down the road, so picking one func to rule them all is probably a good idea. I would like to keep the `SetContent` naming pattern though (to stay in line with Go idioms for getter/setter funcs), so I suppose we should re-name `UpdateContent` to `SetContent`,  remove the old `SetContent`, and do the type assertion within the new `SetContent` to see if we can Add/Update based on the type?

First we should decide on how the content API routes design should or should not change as referenced in the comment [here](https://github.com/ponzu-cms/ponzu/pull/92#discussion_r104325500).